hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestWorkflowImplTest.java

```diff
@@ -0,0 +1,483 @@
+                                        currentPlatformStatus,
+                                        stateAccessor,
+                                        onset,
+                                        checker,
+                                        cryptoService,
+                                        throttleAccumulator,
+                                        null,
+                                        opCounters))
+                .isInstanceOf(NullPointerException.class);
+        assertThatThrownBy(
+                        () ->
+                                new IngestWorkflowImpl(
+                                        nodeInfo,
+                                        currentPlatformStatus,
+                                        stateAccessor,
+                                        onset,
+                                        checker,
+                                        cryptoService,
+                                        throttleAccumulator,
+                                        submissionManager,
+                                        null))
+                .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void testSuccess() throws PreCheckException, InvalidProtocolBufferException {
+        // given
+        final ByteBuffer responseBuffer = ByteBuffer.allocate(1024 * 6);
+
+        // when
+        workflow.submitTransaction(ctx, requestBuffer, responseBuffer);
+
+        // then
+        final TransactionResponse response = parseResponse(responseBuffer);
+        assertThat(response.getNodeTransactionPrecheckCode()).isEqualTo(OK);
+        assertThat(response.getCost()).isZero();
+        verify(opCounters).countReceived(ConsensusCreateTopic);
+        verify(submissionManager).submit(ctx, TRANSACTION_BODY, requestBuffer);
+        verify(opCounters).countSubmitted(ConsensusCreateTopic);
+    }
+
+    @Test
+    void testParseAndCheckWithZeroStakeFails()
+            throws InvalidProtocolBufferException, PreCheckException {
+        // given
+        final ByteBuffer responseBuffer = ByteBuffer.allocate(1024 * 6);
+        when(nodeInfo.isSelfZeroStake()).thenReturn(true);
+
+        // when
+        workflow.submitTransaction(ctx, requestBuffer, responseBuffer);
+
+        // then
+        final TransactionResponse response = parseResponse(responseBuffer);
+        assertThat(response.getNodeTransactionPrecheckCode()).isEqualTo(INVALID_NODE_ACCOUNT);
+        assertThat(response.getCost()).isZero();
+        verify(opCounters, never()).countReceived(any());
+        verify(submissionManager, never()).submit(any(), any(), any());
+        verify(opCounters, never()).countSubmitted(any());
+    }
+
+    @ParameterizedTest
+    @EnumSource(PlatformStatus.class)
+    void testParseAndCheckWithInactivePlatformFails(final PlatformStatus status)
+            throws InvalidProtocolBufferException, PreCheckException {
+        if (status != PlatformStatus.ACTIVE) {
```

Comment:
Should this test pass by default when `status == PlatformStatus.ACTIVE`? I'm confused about the purpose of this 'if' statement

Reply:
It is a parameterized test. It is called with each available status and checks, if the right error condition is returned, if the platform is not active. Therefore the test should not be executed when the `status` is `ACTIVE`.